tortoise/state.go

```diff
@@ -62,8 +62,9 @@ type (
 			min, max types.LayerID
 		}
 
-		epochs map[types.EpochID]*epochInfo
-		layers map[types.LayerID]*layerInfo
+		epochs  map[types.EpochID]*epochInfo
+		layers  map[types.LayerID]*layerInfo
+		ballots map[types.LayerID][]*ballotInfo
```

Comment:
it's definitely worth adding a comments about this recursive behavior.
i think it'll be very tempting for future dev to do the same thing without realizing the subtleties.

Reply:
definitely, added a comment